test/jdk/java/awt/TitledBorder/TitledBorderTest.java

```diff
@@ -0,0 +1,266 @@
+                BufferedImage img = images.get(i);
+                double scaling = scales[i];
+
+                // For vertical count the number of shadow / highlight in
+                // the middle of the image at (x = SIZE.width / 2)
+                // (there must be no background color between these two colours)
+                // Then skip background until the next border where you count
+                // shadow and highlight thickness
+
+                 int x = SIZE.width / 2;
+                 checkVerticalBorder(x, img, scaling);
+
+                // For horizontal border, take the middle of each panel and
+                // count the number of shadow and highlight pixels
+                for (Point p : panelLocations) {
+                    int y = (int) (p.y * scaling) + SIZE.height / 2;
+                    System.out.println(scaling + " : " + y);
+
+                     checkHorizontalBorder(y, img, scaling);
+                }
+        }
+    }
+
+    private static void hBorderLoop(int x1, int x2, int y, BufferedImage buff,
+                                    Color c, double scaling) throws RuntimeException {
+        int thickness = 0;
+        for (int i = x1; i < x2; i++) {
+            if (buff.getRGB(i, y) == c.getRGB()) thickness++;
+        }
+
+        int expected = (int) Math.floor(scaling);
+        if (thickness > expected) {
+            throw new RuntimeException("Horizontal Border drawn too thick. Thickness: "
+                    + thickness + " Scaling: " + scaling + " y: " + y);
+        } else if (thickness < expected) {
+            throw new RuntimeException("Horizontal Border drawn too thin. Thickness: "
+                    + thickness + " Scaling: " + scaling + " y: " + y);
+        }
+    }
+
+    private static void checkHorizontalBorder(int y, BufferedImage img, double scaling) throws RuntimeException {
+        // checking left border
+        hBorderLoop(0, (int) (Math.floor(scaling)+(5*scaling)), y, img, shadow, scaling);
+        hBorderLoop((int) Math.floor(scaling), (int) (Math.floor(scaling)*2+5*scaling), y, img, highlight, scaling);
+
+        // checking right border
+        hBorderLoop(img.getWidth() - ((int) (Math.floor(scaling)*2+5*scaling)), img.getWidth() - ((int) Math.floor(scaling)),
+                y, img, shadow, scaling);
+        hBorderLoop(img.getWidth() - ((int) (Math.floor(scaling)+5*scaling)), img.getWidth(),
+                y, img, highlight, scaling);
+    }
+
+    private static void verifyThickness(int x, int thickness, double scaling) {
+        int expected = (int) Math.floor(scaling);
+        if (thickness < expected) throw new RuntimeException("Vertical Border drawn too thin. Thickness: "
+                + thickness + " Scaling: " + scaling + " x: " + x);
+        if (thickness > expected) throw new RuntimeException("Vertical Border drawn too thick. Thickness: "
+                + thickness + " Scaling: " + scaling + " x: " + x);
```

Comment:
I guess `thickness != expected` is enough.

I'd print both `x` and `y` where the error is detected. On the other hand, it's usually clear where the error is when you look at the image.